include/aspect/vofinterface/vof_system.h

```diff
@@ -0,0 +1,148 @@
+          std::vector<Tensor<1,dim> > face_old_old_velocity_values;
+          std::vector<Tensor<1,dim> > face_mesh_velocity_values;
+
+          std::vector<double>         neighbor_old_values;
+        };
+      }
+
+      namespace CopyData
+      {
+        template <int dim>
+        struct VoFSystem
+        {
+          /**
+           * Constructor.
+           * @param finite_element The element that describes the field for which we
+           *    are trying to assemble a linear system. <b>Not</b> the global finite
+           *    element.
+           */
+          VoFSystem(const FiniteElement<dim> &finite_element);
+          VoFSystem(const VoFSystem &data);
+
+          /**
+           * Local contributions to the global matrix and right hand side
+           * that correspond only to the variables listed in local_dof_indices
+           */
+          FullMatrix<double>          local_matrix;
+          Vector<double>              local_rhs;
+          /**
+           * Local contributions to the global rhs from the face terms in the
+           * discontinuous Galerkin interpretation of the VoF method.  The
+           * vector is of length GeometryInfo<dim>::max_children_per_face *
+           * GeometryInfo<dim>::faces_per_cell so as to hold one matrix for
+           * each possible face or subface of the cell.
+           **/
+          std::vector<Vector<double>>     local_f_rhs;
+          std::vector<FullMatrix<double>> local_f_matrices_ext_ext;
+
+          /**
+           * Denotes which face's rhs have actually been assembled in the DG
+           * field assembly. Entries not used (for example, those corresponding
+           * to non-existent subfaces; or faces being assembled by the
+           * neighboring cell) are set to false.
+           **/
+          std::vector<bool>               assembled_rhs;
+
+          /**
+           * Indices of those degrees of freedom that actually correspond to
+           * the vof field. Since this structure is used to represent just
+           * contributions to the vof systems, there will be no contributions
+           * to other parts of the system and consequently, we do not need to
+           * list here indices that correspond to velocity or pressure degrees
+           * (or, in fact any other variable outside the block we are currently
+           * considering)
+           */
+          std::vector<types::global_dof_index>   local_dof_indices;
+          /**
+           * Indices of the degrees of freedom corresponding to the vof field
+           * on all possible neighboring cells. This is used in the
+           * discontinuous Galerkin interpretation of the VoF method. The outer
+           * std::vector has length GeometryInfo<dim>::max_children_per_face *
+           * GeometryInfo<dim>::faces_per_cell.
+           **/
+          std::vector<std::vector<types::global_dof_index> >   neighbor_dof_indices;
```

Comment:
same here: if you know the size of the array statically, make it a statically sized array